### PR TITLE
New Download Service UI

### PR DESCRIPTION
**Pull request by honeybadgerofdoom**

### CODE REVIEW

File: src/components/NewDataDownload/DatasetArea/DatasetTable.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    },
+    paper: {
+        margin: theme.spacing(2),
+    },
+});
+
+export default function DatasetTable(props: any) {
+    const classes = useStyles();
+    const [granularity, setGranularity] = useState("state");
+    const [filteredDatasets, setFilteredDatasets] = useState(props.data.currentState.datasets);
```
Comment (drg101) [inline src/components/NewDataDownload/DatasetArea/DatasetTable.tsx:85]: type annotations needed on these `useState`

File: src/components/NewDataDownload/DatasetArea/DatasetTable.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                </Paper>
+            </Grid>
+        )
+    }
+
+    else {
+        return null
+    }
+
+    function renderDatasetRows() {
```
Comment (drg101) [inline src/components/NewDataDownload/DatasetArea/DatasetTable.tsx:111]: why is this function below the return statements?

File: src/components/NewDataDownload/DatasetArea/DownloadDatasetPopup.tsx [hunk 3]
```diff
@@ -53,127 +53,159 @@ You may add Your own copyright statement to Your modifications and may provide a
...
+    iconSpacing: {
+        margin: "0px 5px"
+    },
+    headerText: {
+        fontSize: "1em"
+    },
 });
 
-export default React.memo(function DownloadSection() {
+export default function DownloadDatasetPopup(props: any) {
```
Comment (drg101) [inline src/components/NewDataDownload/DatasetArea/DownloadDatasetPopup.tsx:110]: `any` type should be changed

File: src/components/NewDataDownload/DatasetArea/DownloadDatasetPopup.tsx [hunk 4]
```diff
@@ -53,127 +53,159 @@ You may add Your own copyright statement to Your modifications and may provide a
...
-                conductDownload={conductDownload}
-                menumetadata={menumetadata}
-                regionsSorted={regionsSorted}
-                regionGranularity={regionGranularity}
-                setRegionGranularity={setRegionGranularity}
-            />
+    function getTags() {
+        let tags = []
+        if (props.collection.temporal) {
+            tags.push(makeTag("This dataset is temporal, and will have multiple records per entry.", <HourglassEmptyIcon />))
```
Comment (drg101) [inline src/components/NewDataDownload/DatasetArea/DownloadDatasetPopup.tsx:135]: grammar

File: src/components/NewDataDownload/DatasetArea/TableHeader/CountyDropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+import theme from "../../../../global/GlobalTheme";
+
+const useStyles = makeStyles({
+    root: {
+        width: "100%",
+        marginBottom: theme.spacing(1),
+    },
+});
+
+export default function CountyDropdown(props: any) {
```
Comment (drg101) [inline src/components/NewDataDownload/DatasetArea/TableHeader/CountyDropdown.tsx:74]: `any` type

File: src/components/NewDataDownload/DatasetArea/TableHeader/DatasetFilter.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+import theme from "../../../../global/GlobalTheme";
+
+const useStyles = makeStyles({
+    root: {
+        width:"100%",
+        marginTop: theme.spacing(1),
+    },
+});
+
+export default function DatasetFiler(props: any) {
```
Comment (drg101) [inline src/components/NewDataDownload/DatasetArea/TableHeader/DatasetFilter.tsx:73]: `any` type

File: src/components/NewDataDownload/MapArea/Filtering/FilterByStateName.tsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+            color={getColor()}
+            className={props.class}
+            variant="outlined"
+            onChange={handleChange}
+            placeholder="ex: Colorado"
+        />
+    );
+}
+
+function stateMatch(matchString: any, state: any) {
```
Comment (drg101) [inline src/components/NewDataDownload/MapArea/Filtering/FilterByStateName.tsx:90]: `any` types

File: src/components/NewDataDownload/MapArea/Filtering/FilterType.tsx [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import {Select,Box, MenuItem, FormControl} from "@mui/material";
+
+export default function FilterType(props: any) {
```
Comment (drg101) [inline src/components/NewDataDownload/MapArea/Filtering/FilterType.tsx:4]: `any`

File: src/components/NewDataDownload/Utils/utils.tsx [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+    for (let i = 0, x = str.length; i < x; i++) {
+        if (str[i] === null || str[i].length <= 2) {
+            continue;
+        }
+        str[i] = str[i][0].toUpperCase() + str[i].substr(1);
+    }
+    return str.join(" ");
+}
+
+export function lowercaseArray(array: string[]) {
```
Comment (drg101) [inline src/components/NewDataDownload/Utils/utils.tsx:24]: Very nitpicky, but it seems like alot of the for loops in this file should be replaced with `.map` or `for of` loops.

File: src/library/CountyMapping.tsx [hunk 10]
```diff
@@ -0,0 +1,3298 @@
+export const countyMap: { [x: string]: string[]; } = {
+    "Alaska" : [
```
Comment (drg101) [inline src/library/CountyMapping.tsx:2]: This file seems redundant.

File: src/library/d3.min.js [hunk 11]
```diff
@@ -0,0 +1,2 @@
+// https://d3js.org v6.5.0 Copyright 2021 Mike Bostock
```
Comment (drg101) [inline src/library/d3.min.js:1]: Can this go under a /third-party directory?

**Review by drg101 — CHANGES_REQUESTED** [review]
Seems like there are a lot of places where the code quality could be improved with type annotations. Try to avoid using `any` (for the most part, of course some functions will need different treatment).

**Comment by drg101** [comment]
It seems like the collection names are being auto-generated, you should be using the names from the aperture JSON, and then falling back on generating capitalized versions of the names.

**Comment by drg101** [comment]
"Include Geospatial Data" should only be an option on collections which there is a separate collection for the geospatial data (such as neon, census tracts, and counties). Right now it seems to be an option on all datasets.

**Comment by drg101** [comment]
Functionality and looks are very solid though, great job on this. I just want to make sure that people in the future can edit this code fairly easily by including high-quality type annotations.

File: src/components/NewDataDownload/Utils/utils.tsx (hunk 9 above)
Comment (honeybadgerofdoom) [reply src/components/NewDataDownload/Utils/utils.tsx]: Good call - I refactored all of this.

File: src/library/gisInfo.js [hunk 12]
```diff
@@ -0,0 +1,12887 @@
+export const gisStateCounty = {
```
[the remark this answers is not in the document]
Comment (honeybadgerofdoom) [reply src/library/gisInfo.js:1]: We actually do need this file.

File: src/library/CountyMapping.tsx (hunk 10 above)
Comment (honeybadgerofdoom) [reply src/library/CountyMapping.tsx]: Remove this.

File: src/components/NewDataDownload/DatasetArea/DownloadDatasetPopup.tsx (hunk 4 above)
Comment (honeybadgerofdoom) [reply src/components/NewDataDownload/DatasetArea/DownloadDatasetPopup.tsx]: Fixed

File: src/components/NewDataDownload/DatasetArea/DatasetTable.tsx (hunk 2 above)
Comment (honeybadgerofdoom) [reply src/components/NewDataDownload/DatasetArea/DatasetTable.tsx]: Fixed this.

File: src/components/NewDataDownload/DatasetArea/DatasetTable.tsx (hunk 1 above)
Comment (honeybadgerofdoom) [reply src/components/NewDataDownload/DatasetArea/DatasetTable.tsx]: Fixed these.

File: src/library/d3.min.js (hunk 11 above)
Comment (honeybadgerofdoom) [reply src/library/d3.min.js]: Fixed this.

**Comment by honeybadgerofdoom** [comment]
> It seems like the collection names are being auto-generated, you should be using the names from the aperture JSON, and then falling back on generating capitalized versions of the names.

Good idea, I fixed this.

**Comment by honeybadgerofdoom** [comment]
> "Include Geospatial Data" should only be an option on collections which there is a separate collection for the geospatial data (such as neon, census tracts, and counties). Right now it seems to be an option on all datasets.

What am I looking for in the data to make this determination? The fields I'm seeing in the collection object for Neon datasets are collection, fieldMetadata, icon, info, source, linked, temporal. Comparing this to another collection like, say, Trails Geo I only see collection and label.

**Review by drg101 — APPROVED** [review]
everything seems to work

**Pull request merged**